### DISCUSSION TITLE
Klavio integration doesn't work

### DISCUSSION THREAD

**Discussion by bafomet88** [swellstores/community#33] [Help] [answered]
Hello! I pointed this issue in previous community forum, but it remained unresolved. It seems that Your Klavio integration doesn't work, in Klavio I can't see any data sent by swell.

**Comment by dannyleungkc** [comment]
If Klaviyo integration doesn't work for you. I suggest using Omnisend instead. They have the best customer service and I've been using it and it's been great. Though the integration has some bugs that Swell is yet to solve, and Swell don't provide 1 to 1 technical support on a standard plan, if you have any questions about setting up you can ask Omnisend instead and they respond very quick.

**Reply by bafomet88** [reply]
Thank You for advice - i'll definitely try it out!

**Comment by sags95** [comment]
Hey @bafomet88, are you seeing any data in the Klaviyo Activity Feed?

**Reply by bafomet88** [reply]
No, I can"t see any data related to swell. Earlier we tested this integration on "dev" account and it worked for us. The problem started when we opened need, production swell account.

**Comment by vettloffah** [comment]
The klaviyo integration only sends the completed order data to klaviyo in sort of a raw webhook format. It seems pretty limited this way. Here is what I learned setting up our store (still in development):

1. If you want the real time user data sent (e.g. add to cart, begin checkout, etc.) you need to use klaviyo's js library to send the events to the API.
2. Additionally, you need to "identify" the user to klaviyo. You can do this by sending emails from klaviyo which will tag the users, or by using the klaviyo `identify` feature of the api. We are doing this for logged in users. 
3. If you want to make much use of the order data sent by swell, you need to use the klaviyo webhook feature to transform the data, the data doesn't seem to be sent in the format klaviyo understands out of the box. 

Honestly it's probably easier to send the order data using your own webhook than to use the klaviyo integration. 

Klaviyo got an engineer on a call with me to talk through the above, I recommend you reach out to them.

**Comment by rodrigoddalmeida** [comment, marked as answer]
Thanks for the heads up and suggestions - we are considering upgrades to some of our integrations, but no ETA in sight for Klaviyo yet.
